keanu-project/src/main/java/io/improbable/keanu/algorithms/sampling/Prior.java

```diff
@@ -44,14 +46,17 @@ public static NetworkSamples sample(BayesianNetwork bayesNet,
         }
 
         List<? extends Vertex> topologicallySorted = TopologicalSort.sort(bayesNet.getLatentVertices());
+
         Map<VertexId, List> samplesByVertex = new HashMap<>();
+        List<Double> logProbs = new LinkedList<>();
 
         for (int sampleNum = 0; sampleNum < sampleCount; sampleNum++) {
             nextSample(topologicallySorted, random);
             takeSamples(samplesByVertex, fromVertices);
+            logProbs.add(bayesNet.getLogOfMasterP());
         }
 
-        return new NetworkSamples(samplesByVertex, sampleCount);
+        return new NetworkSamples(samplesByVertex, logProbs, sampleCount);
```

Comment:
Looks like Caleb is deleting this file - you might want to check with him if it's worth keeping it around?

Reply:
Yup that should be OK, I just made the change to satisfy the contracts so it doesn't break the tests.

Reply:
Sorry guys! This algo and the RejectionSampler aren't really useful. Prior is replaced by MetropolisHastings and the RejectionSampler only works for a very specific type of problem and even in that case it's effectively a bad version of MetropolisHastings.